### PR TITLE
api(ticdc) add changefeed unit tests and fix bugs

### PR DESCRIPTION
**Pull request by crelax**
### What problem does this PR solve?

Issue Number: close #xxx

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
`None`
```

### CODE REVIEW

File: cdc/api/v1/api.go [hunk 1]
```diff
@@ -263,7 +263,15 @@ func (h *OpenAPI) CreateChangefeed(c *gin.Context) {
 		return
 	}
 
-	up := h.capture.GetUpstreamManager().GetDefaultUpstream()
+	upManager, err := h.capture.GetUpstreamManager()
+	if err != nil {
+		_ = c.Error(err)
```
Comment (liuzix) [inline cdc/api/v1/api.go:268]:
```suggestion
		_ = c.Error(err)
		return
```

File: cdc/api/v1/api.go [hunk 2]
```diff
@@ -263,7 +263,15 @@ func (h *OpenAPI) CreateChangefeed(c *gin.Context) {
...
+	}
+	up, err := upManager.GetDefaultUpstream()
+	if err != nil {
+		_ = c.Error(err)
```
Comment (liuzix) [inline cdc/api/v1/api.go:272]:
```suggestion
		_ = c.Error(err)
		return
```

File: cdc/api/v1/api.go [hunk 3]
```diff
@@ -263,7 +263,15 @@ func (h *OpenAPI) CreateChangefeed(c *gin.Context) {
 		return
 	}
 
-	up := h.capture.GetUpstreamManager().GetDefaultUpstream()
+	upManager, err := h.capture.GetUpstreamManager()
+	if err != nil {
+		_ = c.Error(err)
+	}
+	up, err := upManager.GetDefaultUpstream()
```
Comment (liuzix) [inline cdc/api/v1/api.go:270]: Check for `up == nil`?

File: cdc/api/v2/api_helpers.go [hunk 4]
```diff
@@ -64,8 +65,12 @@ type APIV2Helpers interface {
 	// getPDClient returns a PDClient given the PD cluster addresses and a credential
 	getPDClient(context.Context, []string, *security.Credential) (pd.Client, error)
 
-	// createTiStore wraps the createTiStore method to increase testability
-	createTiStore([]string, *security.Credential) (tidbkv.Storage, error)
+	// callKVCreateTiStore wraps kv.createTiStore method to increase testability
+	callKVCreateTiStore([]string, *security.Credential) (tidbkv.Storage, error)
```
Comment (liuzix) [inline cdc/api/v2/api_helpers.go:69]: - `call...` does not seem a good name.
- Please name the arguments.

File: cdc/api/v2/changefeed.go [hunk 5]
```diff
@@ -42,48 +45,49 @@ const apiOpVarChangefeedID = "changefeed_id"
...
+		}
+		cfg.PDConfig = getUpstreamPDConfig(up)
 	}
 	credential := &security.Credential{
-		CAPath:   config.CAPath,
-		CertPath: config.CertPath,
-		KeyPath:  config.KeyPath,
+		CAPath:   cfg.CAPath,
+		CertPath: cfg.CertPath,
+		KeyPath:  cfg.KeyPath,
```
Comment (liuzix) [inline cdc/api/v2/changefeed.go:65]: Please consider extracting a method for generating the `Credential`.

File: cdc/capture/capture.go [hunk 6]
```diff
@@ -132,8 +133,13 @@ func NewCaptureWithManager4Test(o owner.Owner, m *upstream.Manager) *Capture {
 }
 
 // GetUpstreamManager is a Getter of capture's upstream manager
-func (c *Capture) GetUpstreamManager() *upstream.Manager {
-	return c.upstreamManager
+func (c *Capture) GetUpstreamManager() (*upstream.Manager, error) {
+	c.upstreamManagerMu.RLock()
+	defer c.upstreamManagerMu.RUnlock()
+	if c.upstreamManager == nil {
+		return nil, cerror.ErrUpstreamManagerNotReady
```
Comment (liuzix) [inline cdc/capture/capture.go:140]:
```suggestion
		return nil, cerror.ErrUpstreamManagerNotReady.GenWithStackByArgs()```

File: cdc/api/v1/validator.go [hunk 7]
```diff
@@ -42,7 +42,14 @@ func VerifyCreateChangefeedConfig(
 	capture capture.InfoForAPI,
 ) (*model.ChangeFeedInfo, error) {
 	// TODO(dongmen): we should pass ClusterID in ChangefeedConfig in the upcoming future
```
Comment (asddongmen) [inline cdc/api/v1/validator.go:44]: line 44 should be moved to line 49.

Comment (asddongmen) [inline cdc/api/v1/validator.go:44]: line 44 should be moved to line 49.

File: cdc/api/v2/api_helpers.go [hunk 8]
```diff
@@ -64,8 +65,12 @@ type APIV2Helpers interface {
 	// getPDClient returns a PDClient given the PD cluster addresses and a credential
 	getPDClient(context.Context, []string, *security.Credential) (pd.Client, error)
 
-	// createTiStore wraps the createTiStore method to increase testability
-	createTiStore([]string, *security.Credential) (tidbkv.Storage, error)
+	// getKVCreateTiStore wraps kv.createTiStore method to increase testability
+	getKVTiStore([]string, *security.Credential) (tidbkv.Storage, error)
```
Comment (asddongmen) [inline cdc/api/v2/api_helpers.go:69]: IMO, `getTiStore` is enough.

File: cdc/api/v2/model.go [hunk 9]
```diff
@@ -314,6 +315,19 @@ type RunningError struct {
 	Message string `json:"message"`
 }
 
+// toCredential generates a security.Credential from a PDConfig
+func (cfg *PDConfig) toCredential() *security.Credential {
```
Comment (asddongmen) [inline cdc/api/v2/model.go:319]: How about `getCrendential` ?

**Pull request closed without merging**